service/frontend/workflow_handler.go

```diff
@@ -4273,30 +4284,42 @@ func (wh *WorkflowHandler) getHistoryReverse(
 	nextPageToken []byte,
 	branchToken []byte,
 ) (*historypb.History, []byte, int64, error) {
+
+	// readFullPageEventsReverse reads a full page of history events from history service in reverse orcer. Due to
+	// storage format of V2 History it is not guaranteed that pageSize amount of data is returned. Function returns the
+	// list of history events, the size of data read, the next page token, and an error if present.
```

Comment:
We should clarify what size refers to here in this comment

Reply:
This is copied from `history_manager_util.go` verbatim for now. I find the original implementation (code structure) to be a bit awkward, and in fact sub-optimal w.r.t. `size`. I will volunteer to fix this in a subsequent PR. For now, I can't really come up with a satisfactory comment to describe it, because I think the original author of this code didn't implement it correctly.